odoo/addons/base/models/res_company.py

```diff
@@ -393,3 +394,15 @@ def _all_branches_selected(self):
             len(self.env.companies.root_id) == 1
             and self.env.companies == self.env['res.company'].sudo().search([('id', 'child_of', self.env.company.root_id.ids)])
         )
+
+    def action_res_company_branches_list(self):
```

Comment:
The purpose of this action is to show children companies.  The kind of view is a detail which might be changed later.  So I suggest a more agnostic name like `action_company_branches`.